Add leaderboard api and integrate with frontend

MarcusTXK opened the pull request:
![image](https://github.com/tau-bar/cs4221-leaderboard/assets/50147457/c5c7cf43-2169-4173-a385-57b9e5568b20)


Testing: 
`GET http://localhost:3000/admin/leaderboard?student_id=112455186461203931068&question_id=1`


Edit:
Realised units is wrong, will change to ms tmr

twothicc reviewed (approved):
Would it be possible to format the displayed submission date.
Having T and Z in the output looks ugly.
Z also refers to UTC so the time shown may be confusing to users.

MarcusTXK commented:
oops missed that, thanks for noticing

MarcusTXK commented:
Update time and date
![image](https://github.com/tau-bar/cs4221-leaderboard/assets/50147457/05a4ed3c-f3d9-4ed3-9c0f-4b4776bf5e9e)

Added handling of empty table
![image](https://github.com/tau-bar/cs4221-leaderboard/assets/50147457/f7bb570b-05f7-415c-b0b5-1761c740a216)


Added loading for home page
![image](https://github.com/tau-bar/cs4221-leaderboard/assets/50147457/df959c25-a381-4da3-a3ef-6b05b8c5e582)
